QA-13488: Allowing the visibility of the cut to not be restricted like COPY

XiangYuHe opened the pull request:
## JIRA

https://jira.jahia.org/browse/QA-13488

## Description

Confirmation with @fpral to remove the constraints on CUT to display for "Pages" as well.

**NOTE**: one interesting note is both showOnNodeTypes and hideOnNodeTypes filter the existing items based on the logic not sure if it is mean to be this way.

Commit:
QA-13488 allowing the visibility of the cut to not be restricted like COPY.

gflores-jahia commented:
Clarification: are we showing visibility of both cut and copy now, since we've removed all restrictions? And is this similar behavior to the backported change (I still see hide restriction on the 7.3.8.x change)

XiangYuHe commented:
> Clarification: are we showing visibility of both cut and copy now, since we've removed all restrictions? And is this similar behavior to the backported change (I still see hide restriction on the 7.3.8.x change)

Right we are showing the visibility of both cut and copy instead of just copy.
I am trying to keep the back-port consistent with v8 that is why I added. 
Being said I took an extra look at useNodeChecks.js the showOnNodeTypes seems like is filtering those variable out... (That is why) bad naming perhaps.
